File: Recursion-in-Arrays/Last_Index.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) {
        // write your code here
        Scanner scn = new Scanner(System.in);
        int n = scn.nextInt();
        int arr[] = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = scn.nextInt();
        }
        int x = scn.nextInt();
        System.out.println(lastIndex(arr, 0, x));
    }

    public static int lastIndex(int[] arr, int idx, int x) {
        if (idx == arr.length) {
            return -1;
        }
        int res = lastIndex(arr, idx + 1, x);
        if (res== -1 && arr[idx] == x) {
            return idx;
        } 
        else {
            return res;
        }
    }

}

File: 7. Recursion-with-Arrays/cpp/Last_Index.cpp
#include <iostream>
using namespace std;

int lastIndex(int arr[], int idx, int x, int n){
    // write your code here
    if(idx==n){
        return -1;
    }
    int newIdx=lastIndex(arr,idx+1,x,n);
    if(newIdx==-1){
        if(arr[idx]==x){
            newIdx=idx;
        }
    }
    return newIdx; 

}

int main(){
    int n;
    cin >> n;
    int d;
    int arr[n];
    for (int i = 0; i < n; i++)
        cin >> arr[i];
    cin >> d;
    int p = lastIndex(arr, 0, d, n);
    cout << p << endl;
}